api/src/AppBundle/Entity/Repository/DocumentRepository.php

```diff
@@ -22,4 +25,92 @@ public function retrieveSoftDeleted()
 
         return $records;
     }
+
+    public function getQueuedDocumentsAndSetToInProgress()
+    {
+        $limit = self::ROWS_LIMIT;
+
+        // Using DENSE_RANK here as we get multiple rows for the same document due to multiple report submissions. This
+        // ensures any limit applied will not miss out submissions by chance
```

Comment:
My knowledge here is limited, so more a question for my own curiosity, but does grouping by `document.id` accomplish the same thing? i.e discarding the multiple rows per document?

Reply:
I think that would work in the sense we would have everything grouped together but in this case I'm relying on the additional rows to be able to nest each submission related to a report under the `report_submissions` key.

The alternative was to make a new query to populate these after we had all the queued documents...I'm not totally against this idea as I think it could improve the readability of the query and I'm assuming the difference in performance would be minimal. I'll have a quick try and see.

Reply:
That's fine, thanks for clarifying. I suspected it was because you were needing to hold on to the other rows.